pkg/util/platformworkloadidentity/federatedcredentials.go

```diff
@@ -18,15 +20,18 @@ const (
 )
 
 func GetPlatformWorkloadIdentityFederatedCredName(clusterResourceId, identityResourceId azure.Resource, serviceAccountName string) string {
-	clusterResourceKey := fmt.Sprintf("%s-%s-%s", clusterResourceId.SubscriptionID, clusterResourceId.ResourceGroup, clusterResourceId.ResourceName)
-	name := fmt.Sprintf("%s-%s-%s", clusterResourceKey, serviceAccountName, identityResourceId.ResourceName)
+	sanitizedServiceAccountName := strings.ReplaceAll(serviceAccountName, ":", "-")
+	parts := strings.Split(sanitizedServiceAccountName, "-")
```

Comment:
The unit tests are failing because the tests themselves don't supply a valid service account name for you to sanitize. For example, on line 670 of pkg/cluster/workloadidentityresources_test.go:
```
serviceAccount: "openshift-cloud-controller-manager:cloud-controller-manager",
```
Should be this:
```
serviceAccount: "system:serviceaccount:openshift-cloud-controller-manager:cloud-controller-manager",
```
The same is true on pkg/validate/dynamic/platformworkloadidentityprofile_test.go line 427. This:
```
expectedPlatformIdentity1ExtraFederatedCredName := platformworkloadidentity.GetPlatformWorkloadIdentityFederatedCredName(clusterResourceId, platformIdentity1ResourceId, "something else")
```
Should be a valid service account name like this:
```
expectedPlatformIdentity1ExtraFederatedCredName := platformworkloadidentity.GetPlatformWorkloadIdentityFederatedCredName(clusterResourceId, platformIdentity1ResourceId, "system:serviceaccount:something:else")
```
Hopefully this fixes the unit test issues.